src/nb_enum.cpp

```diff
@@ -43,7 +43,7 @@ PyObject *enum_create(enum_init_data *ed) noexcept {
                 PyUnicode_FromFormat("%U.%U", scope_qualname.ptr(), name.ptr()));
     }
 
-    const char *factory_name = is_arithmetic ? "IntEnum" : "Enum";
+    const char *factory_name = (is_arithmetic || is_flag_enum)  ? (is_flag_enum ? "Flag" : "IntEnum") : "Enum";
```

Comment:
This could be more straightforwardly written as `is_flag_enum ? "Flag" : is_arithmetic ? "IntEnum" : "Enum"`. But I think you might want the combination of flag+arithmetic to give you IntFlag as well.